packages/grid/src/DataBarCellRenderer.ts

```diff
@@ -0,0 +1,615 @@
+ */
+const lerpColor = (color1: Oklab, color2: Oklab, factor: number): Oklab => {
+  const { L: L1, a: a1, b: b1 } = color1;
+  const { L: L2, a: a2, b: b2 } = color2;
+
+  const L = L1 + (L2 - L1) * factor;
+  const a = a1 + (a2 - a1) * factor;
+  const b = b1 + (b2 - b1) * factor;
+
+  return { L, a, b };
+};
+
+class DataBarCellRenderer extends CellRenderer {
+  drawCellContent(
+    context: CanvasRenderingContext2D,
+    state: GridRenderState,
+    column: VisibleIndex,
+    row: VisibleIndex,
+    textOverride?: string
+  ) {
+    const { metrics, model, theme } = state;
+    if (!isDataBarGridModel(model)) {
+      return;
+    }
+    const {
+      modelColumns,
+      modelRows,
+      allRowHeights,
+      allRowYs,
+      firstColumn,
+    } = metrics;
+
+    const isFirstColumn = column === firstColumn;
+    const rowHeight = getOrThrow(allRowHeights, row);
+    const modelRow = getOrThrow(modelRows, row);
+    const modelColumn = getOrThrow(modelColumns, column);
+    const rowY = getOrThrow(allRowYs, row);
+    const textAlign = model.textAlignForCell(modelColumn, modelRow);
+    const text = textOverride ?? model.textForCell(modelColumn, modelRow);
+    const value = Number(text);
+    const { x: textX, y: textY } = this.getTextRenderMetrics(
+      state,
+      column,
+      row
+    );
+    const columnMin = model.minOfColumn(modelColumn);
+    const columnMax = model.maxOfColumn(modelColumn);
+    const axis = model.axisForCell(modelColumn, modelRow);
+    const dataBarColor = model.barColorForCell(modelColumn, modelRow);
+    const valuePlacement = model.valuePlacementForCell(modelColumn, modelRow);
+    const opacity = model.opacityForCell(modelColumn, modelRow);
+    const markers = model.markersForColumn(modelColumn);
+    const hasGradient = model.hasGradientForCell(modelColumn, modelRow);
+    if (columnMin == null || columnMax == null) {
+      return;
+    }
+
+    const {
+      width: maxWidth,
+      x: dataBarX,
+      y: dataBarY,
+      zeroPosition,
+      leftmostPosition,
+      markerXs,
+      totalValueRange,
+      dataBarWidth,
+    } = this.getDataBarRenderMetrics(context, state, column, row);
+
+    context.save();
+    context.textAlign = textAlign;
+    context.fillStyle = dataBarColor;
+    context.textBaseline = 'middle';
+    context.font = theme.font;
+
+    if (valuePlacement !== 'hide') {
+      context.fillText(text, textX, textY);
+    }
+
+    // Draw bar
+    if (hasGradient) {
+      // Draw gradient bar
+
+      const positiveColor = model.positiveColorForCell(modelColumn, modelRow);
+      const negativeColor = model.negativeColorForCell(modelColumn, modelRow);
+      const positiveColorOklab = linearSRGBToOklab(hexToRgb(positiveColor));
+      const negativeColorOklab = linearSRGBToOklab(hexToRgb(negativeColor));
+      const middleColor: Oklab = lerpColor(
+        negativeColorOklab,
+        positiveColorOklab,
+        0.5
+      );
```

Comment:
Double check with @dsmmcken how we should be getting this middle colour, or if it should be defined separately or something. Right now the gradient looks a little washed compared to the examples.

Reply:
I was thinking if the user calls databar with a single positive or negative color, it's not a gradient.

If they want a red-yellow-green, from -100 to 100 gradient they do negative_color=["red", "yellow"],  positive_color: ["yellow, "green"]. There is no "mid" color, each gradient is it's own half. Specify one color get solid, specify multiple, that part is gradient.